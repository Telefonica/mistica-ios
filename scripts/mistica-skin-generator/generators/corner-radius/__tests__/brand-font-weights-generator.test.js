import fs from "fs";
import { resolve } from "path";
import { generateBrandCornerRadius } from "../brand-corner-radius-generator";

test("generateBrandCornerRadius", () => {
  const anyBrandTokensPath = resolve(__dirname, "any-brand-tokens.json");
  const anyBrandTokens = JSON.parse(
    fs.readFileSync(anyBrandTokensPath, "utf-8")
  );
  const brand = { prefix: "O2New" };
  const result = generateBrandCornerRadius(brand, anyBrandTokens);

  const expectedContentPath = resolve(
    __dirname,
    "expected-o2-new-corner-radius.swift"
  );
  const expectedContent = fs.readFileSync(expectedContentPath, "utf-8");
  expect(result).toBe(expectedContent);
});
