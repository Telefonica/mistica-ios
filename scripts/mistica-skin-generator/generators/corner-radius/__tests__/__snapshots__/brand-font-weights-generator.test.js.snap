// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateBrandCornerRadius 1`] = `
"
// Generated using Make
// DO NOT EDIT

import Foundation

struct O2NewCornerRadius: MisticaCornerRadius {
    var avatar: CGFloat = MisticaRadiusConstants.roundedRadius
    var bar: CGFloat = MisticaRadiusConstants.roundedRadius
    var button: CGFloat = MisticaRadiusConstants.roundedRadius
    var checkbox: CGFloat = 2
    var container: CGFloat = 16
    var indicator: CGFloat = MisticaRadiusConstants.roundedRadius
    var input: CGFloat = 12
    var legacyDisplay: CGFloat = 16
    var popup: CGFloat = 8
    var sheet: CGFloat = 16
    var mediaSmall: CGFloat = 8
}
"
`;
